src/sentry/static/sentry/app/components/discover/transactionsList.tsx

```diff
@@ -0,0 +1,370 @@
+    const tableTitles = titles ?? eventView.getFields().map(field => t(field));
+
+    return tableTitles.map((title, index) => (
+      <HeaderCell column={columnOrder[index]} tableMeta={tableMeta} key={index}>
+        {({align}) => {
+          return (
+            <HeadCellContainer>
+              <SortLink
+                align={align}
+                title={title}
+                direction={undefined}
+                canSort={false}
+                generateSortLink={generateSortLink}
+              />
+            </HeadCellContainer>
+          );
+        }}
+      </HeaderCell>
+    ));
+  }
+
+  renderRow(
+    row: TableDataRow,
+    rowIndex: number,
+    columnOrder: TableColumn<React.ReactText>[],
+    tableMeta: MetaType
+  ): React.ReactNode[] {
+    const {organization, location, linkDataTestId, generateFirstLink} = this.props;
+
+    const resultsRow = columnOrder.map((column, index) => {
+      const field = String(column.key);
+      // TODO add a better abstraction for this in fieldRenderers.
+      const fieldName = getAggregateAlias(field);
+      const fieldType = tableMeta[fieldName];
+
+      const fieldRenderer = getFieldRenderer(field, tableMeta);
+      let rendered = fieldRenderer(row, {organization, location});
+
+      if (generateFirstLink) {
+        const isFirstCell = index === 0;
+
+        if (isFirstCell) {
+          const target = generateFirstLink(organization, row, location.query);
+          rendered = (
+            <Link data-test-id={linkDataTestId ?? 'transactions-list-link'} to={target}>
+              {rendered}
+            </Link>
+          );
+        }
+      }
+
+      const isNumeric = ['integer', 'number', 'duration'].includes(fieldType);
+      const key = `${rowIndex}:${column.key}:${index}`;
+
+      return (
+        <BodyCellContainer key={key}>
+          {isNumeric ? (
+            <GridCellNumber>{rendered}</GridCellNumber>
+          ) : (
+            <GridCell>{rendered}</GridCell>
+          )}
+        </BodyCellContainer>
+      );
+    });
+
+    return resultsRow;
+  }
+
+  renderResults() {
+    const {isLoading, tableData} = this.props;
+    let cells: React.ReactNode[] = [];
+
+    if (isLoading) {
+      return cells;
+    }
+    if (!tableData || !tableData.meta || !tableData.data) {
+      return cells;
+    }
+
+    const columnOrder = this.props.eventView.getColumns();
+
+    tableData.data.forEach((row, i: number) => {
+      // Another check to appease tsc
+      if (!tableData.meta) {
+        return;
+      }
+      cells = cells.concat(this.renderRow(row, i, columnOrder, tableData.meta));
+    });
+    return cells;
+  }
+
+  render() {
+    const {isLoading, tableData} = this.props;
+
+    const hasResults =
+      tableData && tableData.data && tableData.meta && tableData.data.length > 0;
+
+    // Custom set the height so we don't have layout shift when results are loaded.
+    const loader = <LoadingIndicator style={{margin: '70px auto'}} />;
```

Comment:
Should the height of the loader be a prop or use the number of results in some way?

Reply:
This approach of setting the margin in the style attribute is used throughout the app. In order to change this to set the size based on a prop will require a decently large sized refactor to ensure we don't break the existing CSS. So I think it'll be better to leave this as is for now.